pkg/sources/adapter/googlecloudpubsubsource/adapter.go

```diff
@@ -101,6 +101,8 @@ func NewAdapter(ctx context.Context, envAcc pkgadapter.EnvConfigAccessor, ceClie
 }
 
 // Start implements adapter.Adapter.
+// Required permissions:
+// - pubsub.subscriptions.consume
```

Comment:
While running the tests manually, I noticed that some required permissions were not documented in the code.
To save other devs the effort of figuring them out, I went ahead and documented them as part of this PR.